mcptest/mcptest_test.go

```diff
@@ -187,3 +187,73 @@ func TestServerWithResource(t *testing.T) {
 		t.Errorf("Got %q, want %q", textContent.Text, want)
 	}
 }
+
+func TestServerWithResourceTemplate(t *testing.T) {
+	ctx := context.Background()
+
+	srv := mcptest.NewUnstartedServer(t)
+	defer srv.Close()
+
+	template := mcp.NewResourceTemplate(
+		"file://users/{userId}/documents/{docId}",
+		"User Document",
+		mcp.WithTemplateDescription("A user's document"),
+		mcp.WithTemplateMIMEType("text/plain"),
+	)
+
+	handler := func(ctx context.Context, request mcp.ReadResourceRequest) ([]mcp.ResourceContents, error) {
+		if request.Params.Arguments == nil {
+			return nil, fmt.Errorf("expected arguments to be populated from URI template")
+		}
+
+		userIds, ok := request.Params.Arguments["userId"].([]string)
+		if !ok {
+			return nil, fmt.Errorf("expected userId argument to be populated from URI template")
+		}
+		if len(userIds) != 1 && userIds[0] != "john" {
```

Comment:
Shoul be `||` instead of `&&` and could be split up for clarity
```
if len(userIds) != 1 { 
  return nil, fmt.Errorf("expected userId to have one value, but got %d", len(userIds)) 
} 
if userIds[0] != "john" { 
  return nil, fmt.Errorf("expected userId argument to be 'john', got %s", userIds[0]) 
}
```

Reply:
@ezynda3 thanks for review, i commit changes https://github.com/mark3labs/mcp-go/pull/449/commits/2f0e9a616b0ce989f327d58331f1449537071d3d